src/db/file.rs

```diff
@@ -63,35 +63,53 @@ pub struct Blob {
 }
 
 pub fn get_path(conn: &Connection, path: &str) -> Option<Blob> {
-    let rows = conn.query("SELECT path, mime, date_updated, content
-                           FROM files
-                           WHERE path = $1", &[&path]).unwrap();
+    if let Some(client) = s3_client() {
```

Comment:
Do you think it would be useful to run the database query anyway, and instead of switching on `content = 'in-s3'`, switch on whether the row exists in the table? I know it makes no difference for the production service, but it feels more resilient for anyone who hosts their own version of it and transitions to S3 after being online for a while.

Reply:
I'd prefer not to, as it does add some amount of overhead and to an extent makes the code more complex. `move-to-s3` should be the preferred route for migrating local installations to S3.

Reply:
Right, i'm just thinking about the period between saving S3 credentials into the environment and the moment `cratesfyi database move-to-s3` finishes. Any calls to the website in that time will fail because we won't be looking at the database for the files.

Reply:
That's true. move-to-s3 is no longer an operation that can run in the background. Since it's likely to never be used again outside local installations though that seems fine -- and if in the future there is a desire to make that happen the code change isn't really that bad.